akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,9 +130,9 @@ object ActorMaterializer {
   /**
    * INTERNAL API
    */
-  private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
-      case m: ActorMaterializer ⇒ m
+  def downcast(materializer: Materializer): ExtendedActorMaterializer =
```

Comment:
I'm fine with opening up
`def downcast(materializer: Materializer): ActorMaterializer`
but `ExtendedActorMaterializer` should not be too visible yet. I think you can add that cast in your own code without problem.